tests/PromiseMiddleware.test.js

```diff
@@ -1,39 +1,32 @@
-import {assert} from 'chai';
+import { spy } from 'sinon';
 import PromiseMiddleware, {START, DONE} from '../src/PromiseMiddleware';
 
-function assertPromiseRejects(promise, assertAgainst) {
-    const matcher = typeof assertAgainst === 'function' ? assertAgainst : value => value === assertAgainst;
-    return promise.then(
-        () => assert(false, 'asserted promise expected to be rejected, but was resolved'),
-        reason => matcher(reason)
-    );
-}
-
 describe('PromiseMiddleware', () => {
     const nextHandler = PromiseMiddleware();
 
     it('must return a function to handle next', () => {
-        assert.isFunction(nextHandler);
-        assert.strictEqual(nextHandler.length, 1);
+        expect(nextHandler).toBeInstanceOf(Function);
+        expect(nextHandler).toHaveLength(1);
     });
 
     describe('handle next', () => {
         it('must return a function to handle action', () => {
             const actionHandler = nextHandler();
 
-            assert.isFunction(actionHandler);
-            assert.strictEqual(actionHandler.length, 1);
+            expect(actionHandler).toBeInstanceOf(Function);
+            expect(actionHandler).toHaveLength(1);
         });
 
         describe('handle action', () => {
-            it('must pass actions without promises to the next handler', done => {
+            it('must pass actions without promises to the next handler', () => {
                 const actionObj = {};
-                const actionHandler = nextHandler(action => {
-                    assert.strictEqual(action, actionObj);
-                    done();
-                });
+                const actionSpy = spy();
+                const actionHandler = nextHandler(actionSpy);
 
                 actionHandler(actionObj);
+
+                expect(actionSpy.calledOnce).toBeTruthy();
+                expect(actionSpy.firstCall.args[0]).toEqual(actionObj);
```

Comment:
I think you can actually drop `sinon` and get away with using Jest's built-in spy, `jest.fn()` -- for example:

```js
const actionSpy = jest.fn();
const actionHandler = nextHandler(actionSpy);

actionHandler(actionObj);

expect(actionSpy).toHaveBeenCalledTimes(1);
expect(actionSpy).toHaveBeenCalledWith(actionObj);
```